net-im/discord/files/launcher.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/env bash
```

Comment:
Unfortunately this is completely broken.

/bin/env may not exist, and if it does, it only exists as an accident of merged-usr. It breaks split-usr (openrc-only) systems.

The portable shebang is `#!/usr/bin/env XXX`. You can also just use `@GENTOO_PORTAGE_EPREFIX@/bin/bash` together with `inherit prefix` and `prefixify_ro`

Reply:
See https://bugs.gentoo.org/935791

Reply:
Sorry for this issue.
Genuinely I never knew the correct path is under /usr, I always used `/bin/env` for 15 years I guess (I learned with Debian, maybe this was the case back in time when boot required binaries were at least all under {/bin,/sbin,/lib,/lib64}?).
Just for information I am under openrc, and when I was with split-usr the binary (or a symlink?) was there.

Reply:
It's a slightly weird situation. Some binaries are guaranteed to be under /bin, but only the ones required for boot -- which doesn't include the `env` program, weirdly. :)

bash, sh, and various other programs canonicalize to /bin, but env canonicalizes to /usr/bin and there are a number of distros that only ship it there. I don't have an openrc gentoo system myself but it is a widespread de-facto standard for the `#!/usr/bin/env progname` shebang to be used when you don't know what directory `progname` can be found in.

Reply:
It’s odd; when I see the coreutils-9.5 ebuild, I see for split-usr USE flag some binaries, `env` included, are moved under /bin and then symlink-ed under /usr/bin (commit id is the commit id closing this PR):
https://github.com/gentoo/gentoo/blob/6368a566a8b05f09e5edfad5d4f7577a83c6334b/sys-apps/coreutils/coreutils-9.5.ebuild#L286-L299

And in case you are under merged-usr profile (like the reporter and I), there is a symlink `/bin → usr/bin`.
Therefore I think the reporter’s system might have a flaw. Otherwise many people would have report the issue on IRC or b.g.o, like it was the case before I sent this PR.

Anyway, sorry for my mistake, I guess I learned a new best-practice.

Reply:
Oh wow, you are right. I didn't notice that the reporter was using merged-usr. :P Then how could /bin not exist???

I also neglected to check whether the ebuild symlinks the binaries on split-usr -- it is possible to do so, but not all distros do, e.g. Alpine Linux definitely does not (they symlink /usr/bin/env -> /bin/busybox so it still depends on /bin but do not provide /bin/env at all).


So it turns out it was safe for regular Gentoo installs all along, but, it is still bad practice for portability. For example, it would break on Alpine Linux when using Gentoo Prefix to install Discord. ;) Hence still a packaging bug but less serious than I initially assumed.